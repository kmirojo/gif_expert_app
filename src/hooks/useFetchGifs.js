import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    // Fires it just when category changes
    useEffect(() => {
        getGifs(category)
            .then((imgs) => {
                setState({
                    data: imgs,
                    loading: false
                });
            })
    }, [category]);

    return state; 
};