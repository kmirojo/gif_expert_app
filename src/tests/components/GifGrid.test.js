import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");



describe('<GifGrid />', () => {
    
    const category = "Man of Steel"

    useFetchGifs.mockReturnValue({
        data: [],
        loading: true,
    });
    const wrapper = shallow(
        <GifGrid category={category} />
    );

    test('should match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when useFetchGifs responds', () => {
        const gifs = [{
            id: "ABC",
            url: "https://localhost/cualquier/cosa.jpg",
            title: "cualquier cosa",
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(
            <GifGrid category={category} />
        );  
        
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find("p").exists() ).toBe(false);
        expect( wrapper.find("GifGridItem")).toHaveLength(gifs.length);
    })
    
    
})
