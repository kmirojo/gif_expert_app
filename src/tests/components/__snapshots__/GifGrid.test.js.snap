// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> should match Snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Man of Steel
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> should show items when useFetchGifs responds 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Man of Steel
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
